### PR TITLE
[Chips] Add client the ability to delete chips in a chip field

### LINKED ISSUE #4787: [Chips] It is difficult to understand how to delete individual chips
From the internal report:

> The core issue is that it is difficult to target delete particular chips in the MDCChipField.
>
> Voice supports display a number selecting controller when user taps on a chip, this overlaps with the proposed tapping on chip to display AlertController.

---

This is an internal issue. If you are a Googler, please visit [b/112262983](http://b/112262983) for more details.

### PR DESCRIPTION
**Pull request by codeman7**
This adds the ability for clients to delete chips within a chip field. 

Within the unit test I couldn't simulate touchUpInside so I felt this was the best option. A client cannot get this behavior unless if they set the flag `showChipsDeleteButton`.

Closes #4787

### CODE REVIEW

**Comment by romoore** [comment]
Please double-check that our internal accessibility team is OK with this change. My most recent understanding is that MDC should not provide tap targets on Chips until we can determine how to meet tap target requirements and provide good UX.

Googlers see go/chips-delete-button-ux

**Comment by romoore** [comment]
To be clear, we don't prevent clients from adding such buttons, just that we don't want to include them in the component until we have solid UX and accessibility recommendations on how to accomplish it well.  Please also be aware that ChipField is unlikely to be useful much longer [because it cannot be easily themed as a Text Field](https://github.com/material-components/material-components-ios/projects/42).

File: components/Chips/src/MDCChipField.h [hunk 1]
```diff
@@ -115,6 +115,15 @@ typedef NS_OPTIONS(NSUInteger, MDCChipFieldDelimiter) {
...
 @property(nonatomic, assign) UIEdgeInsets contentEdgeInsets;
 
+/**
+ Enabling this property allows chips to be deleted by tapping on them.
+
+ @note This does not support the 48x48 touch targets that Google recommends. We recommend if this
+ behavior is enabled that a snackbar or dialog are used as well to allow the user to confirm if they
+ want to delete the chip.
+ */
+@property(nonatomic) BOOL enableChipsThatDelete;
```
Comment (ajsecord) [inline components/Chips/src/MDCChipField.h]: Please rename to `showChipDeleteButton` or similar to match the standard "show*" pattern.

**Review by ajsecord — APPROVED** [review]
I'm ok with this change going in at a high level for the following reasons:

- The Material spec still includes the delete button in its guidance and has no alternative.

- We have—quite reasonably—requests from clients for this functionality.

- The delete button doesn't introduce any further a11y regressions that already exist, since the accessory view is already inaccessible. We should file a bug for making it accessible, of course.

We should definitely keep striving for more-accessible patterns here and when we have them we will update the Material spec and this implementation.

**Pull request merged**